src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -42,4 +43,31 @@ typedef struct {
 
 extern MPIDI_XPMEM_global_t MPIDI_XPMEM_global;
 
+/* ---------------------------------------------------- */
+/* XPMEM auxiliary macros                               */
+/* ---------------------------------------------------- */
+
+#define MPIDI_XPMEM_ACK_TAG 0x8f5f6f7f
+#define MPIDI_XPMEM_HEAD_COUNT 5
+#define MPIDI_NONBLOCKING 0
+#define MPIDI_BLOCKING    1
+
+#define MPIDI_XPMEM_PAGE_SIZE (4096LL)
+#define MPIDI_XPMEM_PAGE_MASK (MPIDI_XPMEM_PAGE_SIZE - 1LL)
+#define MPIDI_XPMEM_PAGE_ALIGN_FLOOR(addr) ((addr) & ~MPIDI_XPMEM_PAGE_MASK)
+#define MPIDI_XPMEM_PAGE_ALIGN_CEIL(addr) ((addr) & MPIDI_XPMEM_PAGE_MASK ? (((addr) & ~MPIDI_XPMEM_PAGE_MASK) + MPIDI_XPMEM_PAGE_SIZE) : (addr))
+
```

Comment:
I have defined `MPL_ROUND_DOWN_ALIGN` and `MPL_ROUND_UP_ALIGN` macros. Let's use them.